src/Streamly/Coreutils/FileTest.hs

```diff
@@ -350,3 +408,87 @@ isOlderThan = compareModTime (<)
 -- Like coreutil @test file1 -ef file2@.
 isSameFile :: FilePath -> IO FileTest
 isSameFile = undefined
+
+getLocalTime :: IO MilliSecond64
+getLocalTime = fromAbsTime <$> getTime Realtime
+
+fileTouchedCompareNminutesBefore ::
+       (MilliSecond64 -> MilliSecond64 -> Bool)
+    -> (FileStatus -> EpochTime)
+    -> Int -> IO FileTest
+fileTouchedCompareNminutesBefore cmp toEpoc n  = do
+    lt <- getLocalTime
+    print lt
+    let diff = MilliSecond64 (fromIntegral n * 60000)
+    return $ FileTest (Predicate (f lt diff))
+
+    where
+
+    f lt diff st =
+         let CTime at = toEpoc st
+             touchedInMsecbefore = (lt - MilliSecond64 (at * 1000))
+           in cmp touchedInMsecbefore diff
+
+fileAccessedLessThanNminutesBefore :: Int -> IO FileTest
+fileAccessedLessThanNminutesBefore =
```

Comment:
Change the name to `isAccessedBefore`, the time should be implicitly in seconds.